Gems/Atom/RHI/DX12/Code/Source/RHI/MemoryView.cpp

```diff
@@ -93,12 +93,22 @@ namespace AZ
         {
             if (m_memoryAllocation.m_memory)
             {
+                // NOTE:  This assumes that name is null-terminated.
                 AZStd::wstring wname;
                 AZStd::to_wstring(wname, name);
                 m_memoryAllocation.m_memory->SetName(wname.data());
             }
         }
 
+        void MemoryView::SetName(const AZStd::wstring_view& name)
```

Comment:
@lumberyard-employee-dm may have different opinions here, but I actually think we should just use `const wchar_t*` to avoid the caveat needed about the string being null terminated unless we are certain we need to support the string view case.

Reply:
Rather than changing the API signature, I'd really prefer to open up a GHI for these methods to switch to using SetPrivateData, since SetName is just doing a strlen() under the covers to get the length and remove the null terminator.

Reply:
I agree with @jeremyong-az  on this one here.
Because the interface is invoking a WinAPI function that expects a LPCWSTR as part of the C interface for ID3D12Resource,
```
HRESULT ( STDMETHODCALLTYPE *SetName )( 
    ID3D12Resource * This,
    _In_z_  LPCWSTR Name);
```

But what is strange is that SetName is not exposed as part of the C++ interface
https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nn-d3d12-id3d12resource

Reply:
It's exposed here:  https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12object-setname .  It's a convenience function on top of this:  https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12object-setprivatedata .  SetPrivateData doesn't take in a null-terminated string.

SetName effectively devolves into a SetPrivateData call similar to this:
(pObject)->SetPrivateData(WKPDID_D3DDebugObjectNameW, wcslen(pName)*2, pName)

Reply:
In that case we could stick with the `wstring_view` implementation and then and invoke SetPrivateData directly then, with `name.size()` and `name.data()`

Reply:
Changed both implementations to use SetPrivateData.

Reply:
Went ahead and switched this to use SetPrivateData.